contrib/gcp/README.md

```diff
@@ -0,0 +1,143 @@
+Portier Deployment to Google Cloud Platform (GCP).
+
+The deployment fronts Portier with Redis running under [Cloud Run](https://cloud.google.com/run) with an [external cross-region global load balancer](https://cloud.google.com/load-balancing/docs/https#cross-region_load_balancing) providing [resilience to region failure](https://cloud.google.com/run/docs/multiple-regions).
+
+Costs are ~$50/region/month plus ~$20/month when idle:
+
+ * [Cloud Run](https://cloud.google.com/run/pricing#tables) (~$10/region/month)
+     * You may wish to pick a Tier 1 region for better pricing where possible
+ * [Redis](https://cloud.google.com/memorystore/docs/redis/pricing#instance_pricing) (~$40/region/month)
+     * Uses basic tier with 1GiB capacity
+ * [Load Balancer](https://cloud.google.com/vpc/network-pricing#lb) (~$20/month)
+     * Deployment fits within the first five forwarding rules pricing
+
+It is recommended, though not necessary, that you deploy to two regions for the purpose of resilience. Further adding more regions is unlikely to provide any measurable benefits as low latency is not a requirement of a service such as Portier.
+
+## Related Links
+
+ * [Deployment Manager](https://cloud.google.com/deployment-manager)
+ * [Jinja Template Documentation](https://jinja.palletsprojects.com/en/2.11.x/templates/)
+
+## Outstanding Issues
+
+ * The [type providers](https://cloud.google.com/deployment-manager/docs/configuration/type-providers/advanced-configuration-options) used need massaging to allow updates
+ * Hard coding of the Redis port to `6379/tcp`
+     * workaround: `Reference [$(ref.portier-europe-west4-redis.port)], was not of type string but [NUMBER], cannot replace inside ["redis://:abcd@192.0.2.1:$($(ref.portier-europe-west4-redis.port))/0\n"]`
+     * GCP's Deployment Manager is mostly awful, so when GCP throws you lemons, it provides zero tools (or documentation) to make lemonade
+         * there is no cast operator available that works on `$(ref ...)`
+         * we cannot use [`outputs`](https://cloud.google.com/deployment-manager/docs/configuration/expose-information-outputs) as it uses pass by reference and creates the same problem
+ * support for [in-transit encryption](https://cloud.google.com/memorystore/docs/redis/in-transit-encryption)
+     * [redis crate supports it](https://docs.rs/redis/0.20.0/redis/enum.ConnectionAddr.html#variant.TcpTls) though Portier's [`pubsub.rs`](../../src/utils/redis/pubsub.rs) explicitly does not
```

Comment:
Ah, sorry about that. I'm not super happy with the pubsub code, but also haven't really worked with upstream to get us everything we need. Maybe newer versions already do what we want. (Can't say, don't remember off the top of my head what the exact pain points were.)